src/aleph/web/controllers/aggregates.py

```diff
@@ -57,17 +60,42 @@ async def address_aggregate(request: web.Request) -> web.Response:
             refresh_aggregate(session=session, owner=address, key=key)
             session.commit()
 
-        aggregates: List[Tuple[str, Dict[str, Any]]] = list(
+        aggregates = list(
             get_aggregates_by_owner(
-                session=session, owner=address, keys=query_params.keys
+                session=session, owner=address, with_info=query_params.with_info
             )
         )
 
-    if not aggregates:
-        return web.HTTPNotFound(text="No aggregate found for this address")
+        if not aggregates:
+            return web.HTTPNotFound(text="No aggregate found for this address")
 
     output = {
         "address": address,
         "data": {result[0]: result[1] for result in aggregates},
     }
+    info = {}
+    if query_params.with_info:
+        for result in aggregates:
+            (
+                aggregate_key,
+                content,
+                created,
+                last_updated,
+                original_item_hash,
+                last_update_item_hash,
+            ) = result
+
+            if isinstance(created, dt.datetime):
+                created = created.isoformat()
+            if isinstance(last_updated, dt.datetime):
+                last_updated = last_updated.isoformat()
+
+            info[aggregate_key] = {
+                "created": created,
+                "last_updated": last_updated,
+                "original_item_hash": original_item_hash,
+                "last_update_item_hash": last_update_item_hash,
+            }
+        output["info"] = info
```

Comment:
The way you're doing this, you're iterating twice over the `aggregates` list. The output dict can be generated with only one iteration.